openraft/src/core/install_snapshot.rs

```diff
@@ -260,13 +260,15 @@ impl<C: RaftTypeConfig, N: RaftNetworkFactory<C>, S: RaftStorage<C>> RaftCore<C,
 
         // A local log that is <= last_applied may be inconsistent with the leader.
         // It has to purge all of them to prevent these log form being replicated, when this node becomes leader.
+        self.engine.update_snapshot_last_log(Some(last_applied));
         self.engine.purge_log(last_applied);
         self.run_engine_commands::<Entry<C>>(&[]).await?;
 
         self.engine.update_committed_membership(req.meta.last_membership);
         self.run_engine_commands::<Entry<C>>(&[]).await?;
 
         self.snapshot_last_log_id = self.engine.state.last_applied;
+        self.engine.update_snapshot_last_log(self.snapshot_last_log_id);
```

Comment:
There should be only one `snapshot_last_log_id`, in either `RaftCore` or `Engine`.

`RaftCore.snapshot_last_log_id` can be removed.

Reply:
Sure, removed.